pin/gc/gc.go

```diff
@@ -17,18 +18,19 @@ var log = logging.Logger("gc")
 // GC performs a mark and sweep garbage collection of the blocks in the blockstore
 // first, it creates a 'marked' set and adds to it the following:
 // - all recursively pinned blocks, plus all of their descendants (recursively)
+// - the root directory for the files API, plus all of its descendants (recursively)
 // - all directly pinned blocks
 // - all blocks utilized internally by the pinner
 //
 // The routine then iterates over every block in the blockstore and
 // deletes any block that is not found in the marked set.
-func GC(ctx context.Context, bs bstore.GCBlockstore, pn pin.Pinner) (<-chan key.Key, error) {
+func GC(ctx context.Context, bs bstore.GCBlockstore, pn pin.Pinner, filesRoot *mfs.Root) (<-chan key.Key, error) {
```

Comment:
instead of passing in an mfs.Root here, i think the better option is just to pass in a keyset. That way we keep it generic and don't build weird dependencies between packages like this